src/cim/components/modals/CimConfiguration/utils.ts

```diff
@@ -0,0 +1,64 @@
+import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+
+import { AgentServiceConfigConditionType, AgentServiceConfigK8sResource } from '../../../types';
+import { getConditionsByType } from '../../../utils';
+import { CIM_CONFIG_TIMEOUT } from './constants';
+
+export const LOCAL_STORAGE_ID_LAST_UPDATE_TIMESTAMP = 'cim-config-configuring-started-at';
+
+export const isCIMEnabled = ({
+  agentServiceConfig,
+}: {
+  agentServiceConfig?: AgentServiceConfigK8sResource;
+}): boolean => !!agentServiceConfig;
+
+export const isCIMConfigured = ({
+  agentServiceConfig,
+}: {
+  agentServiceConfig?: AgentServiceConfigK8sResource;
+}): boolean => {
+  const deploymentsHealthyCondition = getConditionsByType<AgentServiceConfigConditionType>(
+    agentServiceConfig?.status?.conditions || [],
+    'DeploymentsHealthy',
+  )?.[0];
+
+  return deploymentsHealthyCondition?.status === 'True';
+};
+
+export const isCIMConfigProgressing = ({
+  agentServiceConfig,
+}: {
+  agentServiceConfig?: AgentServiceConfigK8sResource;
+}): boolean => {
+  if (!agentServiceConfig) {
+    return false;
+  }
+
+  const deploymentsHealthyCondition = getConditionsByType<AgentServiceConfigConditionType>(
+    agentServiceConfig?.status?.conditions || [],
+    'DeploymentsHealthy',
+  )?.[0];
+
+  const now = Date.now();
+
+  // Potential post-creation change of the ingress controller (from No to Yes) forces the AI deployment reconciliation and so reporting "errors" during the course.
+  // For that reason we can not reuse AgentServiceController's creationTimestamp to report issues with a graceful delay.
+  // We can store timestamp into a label or simply to the localStorage - let's see if that is an issue in a multi-tenant environment.
```

Comment:
cant we somehow use `lastTransitionTime`  of `DeploymentsHealthy` condition ?

Reply:
Maybe. But we would not be able to implement the delay "since user hits the Configure button"-action.